pkg/proxy/unidler/signaler.go

```diff
@@ -0,0 +1,49 @@
+package unidler
+
+import (
+	"fmt"
+
+	idlingclient "github.com/openshift/service-idler/pkg/client/clientset/versioned/typed/idling/v1alpha2"
+	"k8s.io/apimachinery/pkg/types"
+
+	idlingutil "github.com/openshift/origin/pkg/idling"
+)
+
+type NeedPodsSignaler interface {
+	// NeedPods signals that endpoint addresses are needed in order to
+	// service a traffic coming to the given service and port
+	NeedPods(serviceName types.NamespacedName) error
+}
+
+type idlerSignaler struct {
+	idlerClient idlingclient.IdlersGetter
+	lookup      idlingutil.IdlerServiceLookup
+}
+
+func (s *idlerSignaler) NeedPods(service types.NamespacedName) error {
+	idler, present, err := s.lookup.IdlerByService(service)
+	if err != nil {
+		return err
+	}
+	if !present {
+		return fmt.Errorf("no idler found for service %s", service.String())
+	}
+
+	if idler.Spec.WantIdle == false {
```

Comment:
I haven't reviewed the API design yet, but unless this becomes a constant you should never be comparing to a boolean, just testing.

Reply:
whoops, agreed -- must have been tired when I wrote that ;-)